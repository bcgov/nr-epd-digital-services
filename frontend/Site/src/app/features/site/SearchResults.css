.results-loading
{
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 5px;
}




@media (max-width: 768px)
{
    .hide-custom
    {
        display: none;
    }
}