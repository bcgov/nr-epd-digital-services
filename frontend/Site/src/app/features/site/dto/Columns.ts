import { FormFieldType } from "../../../components/input-controls/IFormField";
import { TableColumn , ColumnSize } from "../../../components/table/TableColumn";

const getSiteSearchResultsColumns = () => {
  const columns: TableColumn[] = [
    
    new TableColumn(
      1,
      "Site ID",
      true,
      "id",
      1,
      true,
      true,
      1,
      true,
      getColumnType("Site ID","id","")    
    ),
    new TableColumn(
      6,
      "Site Address",
      true,
      "addrLine_1,addrLine_2,addrLine_3",
      2,
      true,
      true,
      1,
      true,
      getColumnType("Site Address","addrLine_1,addrLine_2,addrLine_3","")  ,
      "site/details/",
      false,
      ColumnSize.Triple
      ,
      
      
    ),
    new TableColumn(
      7,
      "City",
      true,
      "city",
      2,
      false,
      true,
      1,
      true,
      getColumnType("City","city","")  
    ),
    new TableColumn(
      2,
      "Site Remediation Status",
      true,
      "srStatus",
      1,
      false,
      true,
      1,
      false,
      getColumnType("Site Remediation Status","srStatus","")  
    ),
    new TableColumn(
      3,
      "Site Risk code",
      true,
      "siteRiskCode",
      1,
      false,
      true,
      1,
      false,
      getColumnType("Site Risk code","siteRiskCode","")  
    ),
    new TableColumn(
      4,
      "General Description",
      true,
      "generalDescription",
      1,
      false,
      true,
      1,
      false,
      getColumnType("General Description","generalDescription","")  
    ),
    new TableColumn(
      5,
      "Common Name",
      true,
      "commonName",
      2,
      false,
      true,
      1,
      false,
      getColumnType("Common Name","commonName","")  
    ),
    new TableColumn(
      8,
      "Latitude",
      true,
      "latdeg",
      2,
      false,
      true,
      1,
      false,
      getColumnType("Latitude","latdeg","")  
    ),
    new TableColumn(
      9,
      "Latitude(D,M,S)",
      true,
      "latDegrees,latMinutes,latSeconds",
      3,
      false,
      true,
      1,
      false,
      getColumnType("Latitude(D,M,S)","latDegrees,latMinutes,latSeconds","")  
    ),
    new TableColumn(
      10,
      "Longitude",
      true,
      "longdeg",
      3,
      false,
      true,
      1,
      false,
      getColumnType("Longitude","longdeg","")  
    ),
    new TableColumn(
      11,
      "Longitude(D,M,S)",
      true,
      "longDegrees,longMinutes,longSeconds",
      3,
      false,
      true,
      1,
      false,
      getColumnType("Longitude(D,M,S)","longDegrees,longMinutes,longSeconds","") 
    ),
    new TableColumn(
      12,
      "Lat/Long Reliability",
      true,
      "latlongReliabilityFlag",
      3,
      false,
      true,
      1,
      false,
      getColumnType("Lat/Long Reliability","latlongReliabilityFlag","") 
    ),
    new TableColumn(
      13,
      "Created By",
      true,
      "whoCreated",
      4,
      false,
      true,
      1,
      false,
      getColumnType("Created By","whoCreated","") 
    ),
    new TableColumn(
      14,
      "Date Created",
      true,
      "whenCreated",
      4,
      false,
      true,
      1,
      false,
      getColumnType("Date Created","whenCreated","") 
    ),
    new TableColumn(
      15,
      "Last Updated",
      true,
      "whenCreated",
      4,
      false,
      true,
      1,
      false,
      getColumnType("Last Updated","whenCreated","") 
    ),
    new TableColumn(
      16,
      "Consultant Submitted",
      true,
      "consultantSubmitted",
      4,
      false,
      true,
      1,
      false,
      getColumnType("Consultant Submitted","consultantSubmitted","") 
    ),
    new TableColumn(
      17,
      "View",
      true,
      "id",
      4,
      true,
      true,
      1,
      true,
      getLinkColumnType("Map","id","","site/map/") ,
      "site/map/",
      true
    ),
    new TableColumn(
      18,
      "Details",
      true,
      "id",
      4,
      true,
      true,
      1,
      true,
      getLinkColumnType("Details","id","","site/details/"),
      "site/details/",
    
      true,      
    ),
  ];

  return columns;
};


const getColumnType = (label:string, propertyName:string, value:string) =>
{
  return   {
    type: FormFieldType.Label,
    label: label,       
    graphQLPropertyName: propertyName,
    value:value,      
    customLabelCss: "custom-lbl-text",
    customInputTextCss: "custom-input-text",
    tableMode: true,
  }
}

const getLinkColumnType = (label:string, propertyName:string, value:string, href:string) =>
  {
    return   {
      type: FormFieldType.Link,
      label: label,       
      graphQLPropertyName: propertyName,
      value:value,      
      customLabelCss: "link-for-table",
      customInputTextCss: "link-for-table",
      tableMode: true,
      href: href
    }
  }

export { getSiteSearchResultsColumns };

export const B: any =1 ;


