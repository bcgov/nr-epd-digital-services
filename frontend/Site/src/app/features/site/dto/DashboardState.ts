import { RequestStatus } from "../../../helpers/requests/status";

interface IDashboard {
    recentView: []
    recentFolios: []
    recentAssigned: []
}

export interface IDashboardState {
    dashboard: IDashboard;
    error?: string;
    status: string;
}
