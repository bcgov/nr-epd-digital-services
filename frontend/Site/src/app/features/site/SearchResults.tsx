import React, { useEffect } from "react";
import { SpinnerIcon, SortIcon } from "../../components/common/icon";
import "./SearchResults.css";
import { loadingState } from "./dto/SiteSlice";
import { RequestStatus } from "../../helpers/requests/status";
import { useSelector } from "react-redux";

const SearchResults = ({ data }: { data: any[] }) => {
  const columns: string[] = [
    "Site ID",
    "Site Address",
    "City",
    "Region",
    "Last Updated Date",
    "Map",
    "Details",
  ];

  const requestStatus = useSelector(loadingState);

  useEffect(() => {
    console.log("loadingState", requestStatus);
  }, [requestStatus]);

  return (
    <table className="table" aria-label="Search Results">
      <thead aria-label="Search Results Header" >
        <tr className="search-results-section-header">
          <th scope="col" className="search-results-th">
            <input type="checkbox" className="checkbox-color" />
          </th>
          {/* {columns.map((item, index) => {            
            if (item === 'Region' || item === 'Last Updated Date') {
              return (
                <th key={index} scope="col" className="search-results-th hide-custom">
                  {item}
                  <SortIcon className="search-results-sort-icon" />
                </th>
              );
            } else {
              return (
                <th key={index} scope="col" className="search-results-th">
                  {item}
                  <SortIcon className="search-results-sort-icon" />
                </th>
              );
            }
          })} */}

{columns.map((item, index) => (
            <th key={index} scope="col" className={`search-results-th ${item === 'Region' || item === 'Last Updated Date' ? 'hide-custom' : ''}`}>
              {item}
              <SortIcon className="search-results-sort-icon" />
            </th>
          ))}
        </tr>
      </thead>
      <tbody>
        {data.length === 0 ? (
          <tr>
            <td colSpan={8} className="noContent border-quick-color">
              {requestStatus === RequestStatus.loading ? (
                <div className="results-loading">
                  <SpinnerIcon
                    data-testid="loading-spinner"
                    className="fa-spin "
                  />
                  <span className="noContentText"> Searching </span>
                </div>
              ) : (
                <span className="noContentText">No Results Found</span>
              )}
            </td>
          </tr>
        ) : (
          data.map((item: any) => (
            <tr key={item.id}>
              <td className="border-quick-color search-results-text">
                <input
                  type="checkbox"
                  className="checkbox-color"
                  aria-label="Select Row"
                />
              </td>
              <td className="border-quick-color search-results-text">
              
                <a href="#" aria-label={`Site ID ${item.siteId}`}>
                  {item.id}
                </a>
              </td>
              <td
                className="border-quick-color search-results-text"
                aria-label={`Site Address ${item.address}`}
              >
                {item.addrLine_1 +
                  " " +
                  item.addrLine_2 +
                  " " +
                  item.addrLine_3}
              </td>
              <td
                className="border-quick-color search-results-text"
                aria-label={`City ${item.city}`}
              >
                {item.city}
              </td>
              <td
                className="border-quick-color search-results-text hide-custom"
                aria-label={`Region ${item.provState}`}
              >
                {item.provState}
              </td>
              <td
                className="border-quick-color search-results-text hide-custom"
                aria-label={`Region ${item.whenCreated}`}
              >
                {item.whenCreated}
              </td>
              <td className="border-quick-color search-results-text">
                <a href="/map">View</a>
              </td>
              <td className="border-quick-color search-results-text">
                <a href="/site/details">Details</a>
              </td>
            </tr>
          ))
        )}
      </tbody>
    </table>
  );
};

export default SearchResults;
