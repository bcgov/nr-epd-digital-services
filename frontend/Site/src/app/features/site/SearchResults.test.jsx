import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchResults from './SearchResults';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; 
import { RequestStatus } from '../../helpers/requests/status';
import { getSiteSearchResultsColumns } from "./dto/Columns";

const mockStore = configureStore([]);

describe('SearchResults Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      sites: [],
      error: '',
      fetchStatus: RequestStatus.loading,
      deleteStatus: RequestStatus.idle,
      addedStatus: RequestStatus.idle,
      updateStatus: RequestStatus.idle
    });
  });

  test('renders no results found when data is empty', () => {
    const emptyData = [];
    const { container } = render(<Provider store={store}><SearchResults data={emptyData} pageChange={()=>{}} /></Provider>);
    const noResultsText = screen.getByText('No Results Found');
    expect(noResultsText).toBeInTheDocument();
  
  });

  test('renders table rows with data', () => {
    const mockData = [
      {
        siteId: 1,
        id: 'site1',
        address: '123 Main St',
        city: 'Cityville',
        provState: 'State',
        whenCreated: '2024-04-04'
      },   
    ];
    const { container } = render(<Provider store={store}><SearchResults data={mockData} pageChange={(currentPage,resultsPerPage)=>{}} /></Provider>);
    const siteIdLink = screen.getByText('View');
    expect(siteIdLink).toBeInTheDocument();
  });

  test('checkbox selects row when clicked', () => {
    const mockData = [
      {
        siteId: 1,
        id: 'site1',
        address: '123 Main St',
        city: 'Cityville',
        provState: 'State',
        whenCreated: '2024-04-04'
      },    
    ];
    render(<Provider store={store}><SearchResults data={mockData} pageChange={(currentPage,resultsPerPage)=>{}} /></Provider>);
    const checkbox = screen.getByLabelText('Select Row');
    expect(checkbox).toBeInTheDocument();
    userEvent.click(checkbox);
    expect(checkbox).toBeChecked();
  });

  test('renders with no columns provided', () => {

    const columns = getSiteSearchResultsColumns();

    const mockData = [
      {
        siteId: 1,
        id: 'site1',
        address: '123 Main St',
        city: 'Cityville',
        provState: 'State',
        whenCreated: '2024-04-04'
      },
   
    ];
    render(<Provider store={store}><SearchResults data={mockData} columns={columns} pageChange={()=>{}} /></Provider>);
    const siteIdLink = screen.getByText('View');
    expect(siteIdLink).toBeInTheDocument();
  });
});
