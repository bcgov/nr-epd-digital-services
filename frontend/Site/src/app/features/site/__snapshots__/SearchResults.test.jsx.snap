// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Component renders loading state when data is loading 1`] = `
<div>
  <table
    aria-label="Search Results"
    class="table"
  >
    <thead>
      <tr
        class="search-results-section-header"
      >
        <th
          class="search-results-th"
          scope="col"
        >
          <input
            class="checkbox-color"
            type="checkbox"
          />
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Site ID
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Site Address
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          City
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Region
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
           Last Updated Date
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Map
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Details
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="noContent border-quick-color"
          colspan="8"
        >
          <span
            class="noContentText"
          >
            No Results Found
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`SearchResults Component renders no results found when data is empty 1`] = `
<div>
  <table
    aria-label="Search Results"
    class="table"
  >
    <thead>
      <tr
        class="search-results-section-header"
      >
        <th
          class="search-results-th"
          scope="col"
        >
          <input
            class="checkbox-color"
            type="checkbox"
          />
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Site ID
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Site Address
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          City
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Region
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
           Last Updated Date
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Map
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Details
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="noContent border-quick-color"
          colspan="8"
        >
          <span
            class="noContentText"
          >
            No Results Found
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`SearchResults Component renders table rows with data 1`] = `
<div>
  <table
    aria-label="Search Results"
    class="table"
  >
    <thead>
      <tr
        class="search-results-section-header"
      >
        <th
          class="search-results-th"
          scope="col"
        >
          <input
            class="checkbox-color"
            type="checkbox"
          />
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Site ID
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Site Address
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          City
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Region
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
           Last Updated Date
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Map
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
        <th
          class="search-results-th "
          scope="col"
        >
          Details
          <svg
            class="search-results-sort-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.4 41.4c12.5-12.5 32.8-12.5 45.3 0l128 128c9.2 9.2 11.9 22.9 6.9 34.9s-16.6 19.8-29.6 19.8H32c-12.9 0-24.6-7.8-29.6-19.8s-2.2-25.7 6.9-34.9l128-128zm0 429.3l-128-128c-9.2-9.2-11.9-22.9-6.9-34.9s16.6-19.8 29.6-19.8H288c12.9 0 24.6 7.8 29.6 19.8s2.2 25.7-6.9 34.9l-128 128c-12.5 12.5-32.8 12.5-45.3 0z"
            />
          </svg>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="border-quick-color search-results-text"
        >
          <input
            aria-label="Select Row"
            class="checkbox-color"
            type="checkbox"
          />
        </td>
        <td
          class="border-quick-color search-results-text"
        >
           
          <a
            aria-label="Site ID 1"
            href="#"
          >
            site1
          </a>
        </td>
        <td
          aria-label="Site Address 123 Main St"
          class="border-quick-color search-results-text"
        >
          undefined undefined undefined
        </td>
        <td
          aria-label="City Cityville"
          class="border-quick-color search-results-text"
        >
          Cityville
        </td>
        <td
          aria-label="Region State"
          class="border-quick-color search-results-text"
        >
          State
        </td>
        <td
          aria-label="Region 2024-04-04"
          class="border-quick-color search-results-text"
        >
          2024-04-04
        </td>
        <td
          class="border-quick-color search-results-text"
        >
          <a
            href="#"
          >
            View
          </a>
        </td>
        <td
          class="border-quick-color search-results-text"
        >
          <a
            href="#"
          >
            Details
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
